tests/signing.py

```diff
@@ -0,0 +1,198 @@
+    msg.key = cose_key
+
+    return msg.encode()
+
+
+def get_cert_key_type(cert_pem: str) -> str:
+    cert = load_pem_x509_certificate(cert_pem.encode("ascii"), default_backend())
+    if isinstance(cert.public_key(), EllipticCurvePublicKey):
+        return "ec"
+    raise NotImplementedError("unsupported key type")
+
+
+def verify_cose_sign1(buf: bytes, cert_pem: str):
+    key_type = get_cert_key_type(cert_pem)
+    cert = load_pem_x509_certificate(cert_pem.encode("ascii"), default_backend())
+    key = cert.public_key()
+    if key_type == "ec":
+        cose_key = from_cryptography_eckey_obj(key)
+    else:
+        raise NotImplementedError("unsupported key type")
+    msg = Sign1Message.decode(buf)
+    msg.key = cose_key
+    if not msg.verify_signature():
+        raise ValueError("signature is invalid")
+    return msg
+
+
+def detach_content(msg: bytes):
+    m = cbor2.loads(msg)
+    content = m.value[2]
+    m.value[2] = None
+    return content, cbor2.dumps(m)
+
+
+def attach_content(content, detached_envelope):
+    m = cbor2.loads(detached_envelope)
+    m.value[2] = content
+    return cbor2.dumps(m)
+
+
+PRIV = """-----BEGIN EC PARAMETERS-----
+BgUrgQQAIg==
+-----END EC PARAMETERS-----
+-----BEGIN EC PRIVATE KEY-----
+MIGkAgEBBDDMwIszb3ZmKpeHq/vPoz6qnxheI89T2IZpKFQHJwQrvuaFFLDUKK9Z
+jKRMshAeALagBwYFK4EEACKhZANiAAQ38JreTF2uKVaTKBd7fAkIy2bg5U6T0O+H
+wcxJOLgqK+fwidnVlPG+GQUwIj6ik7Xp/0Ig7RVSAyAjcpYWL4dHU5gJ/g9PruHz
+cnmFtP88dARPH2EKy0n/iGh9yXD3bXw=
+-----END EC PRIVATE KEY-----
+"""
+
+PUB = """-----BEGIN CERTIFICATE-----
+MIIBtjCCATygAwIBAgIUJCUauYlNsJ76zOUomey4cF7F+pUwCgYIKoZIzj0EAwMw
+EjEQMA4GA1UEAwwHbWVtYmVyMDAeFw0yMjA5MDYxMzQ2NDlaFw0yMzA5MDYxMzQ2
+NDlaMBIxEDAOBgNVBAMMB21lbWJlcjAwdjAQBgcqhkjOPQIBBgUrgQQAIgNiAAQ3
+8JreTF2uKVaTKBd7fAkIy2bg5U6T0O+HwcxJOLgqK+fwidnVlPG+GQUwIj6ik7Xp
+/0Ig7RVSAyAjcpYWL4dHU5gJ/g9PruHzcnmFtP88dARPH2EKy0n/iGh9yXD3bXyj
+UzBRMB0GA1UdDgQWBBTpme2NGI1y3OY8XYT5XwcuuvG55jAfBgNVHSMEGDAWgBTp
+me2NGI1y3OY8XYT5XwcuuvG55jAPBgNVHRMBAf8EBTADAQH/MAoGCCqGSM49BAMD
+A2gAMGUCMDg1QddcE5YFrcHmFvyXW2s7LaV0NYx24lwImrgWXQTOv7iNXAfrogzP
+CQxyHqkSxgIxANmkmLCojf5NCvwxI5tf37i6zGQ0c9zR0P9b4FtcznEtrbzmXfdJ
+b2H04E57XZmVdg==
+-----END CERTIFICATE-----
+"""
+
+if __name__ == "__main__":
+    signed_statement = create_cose_sign1(
+        b"governance js here", PRIV, PUB, {"ccf_governance_action": "proposal"}
+    )
+    msg = verify_cose_sign1(signed_statement, PUB)
+    assert msg.phdr[cose.headers.KID] == cert_fingerprint(PUB), (
+        msg.phdr[cose.headers.KID],
+        cert_fingerprint(PUB),
+    )
+    content, detached_envelope = detach_content(signed_statement)
+    signed_statement = attach_content(content, detached_envelope)
+    msg = verify_cose_sign1(signed_statement, PUB)
+    signed_statement = create_cose_sign1(
+        b"governance js here",
+        PRIV,
+        PUB,
+        {"ccf_governance_action": "proposal", "ccf_governance_proposal_id": "12345"},
+    )
```

Comment:
Looks like this was purely used for testing - can we remove it now?

Reply:
I'd like to keep it until I've done the detaching in the verifier.